File: Day4/Autolayout Homework Objective C/Autolayout Homework Objective C/ViewController.m
//
//  ViewController.m
//  Autolayout Homework Objective C
//
//  Created by Consultant on 8/23/22.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

//brackets are simile of dot syntax
//foo = [myObject backgroundColor];
//foo = myObject.backgroundColor;

//pass message (call func) by :
//[dateFormatter setDateFormat:@"MM-dd-yyyy"]
//dateFormatter.setDateFormat(@"MM-dd-yyyy")

//strings:
//NSString* flavor = @"Chocolate"
//@"Chocolate" is just a shortcut to create an NSString instance. You could also use:
//[[NSString alloc] initWithUTF8String:"Chocolate"];

- (void)viewDidLoad {
    NSDateFormatter *dateFormatter=[[NSDateFormatter alloc] init];
    [dateFormatter setDateFormat:@"MM-dd-yyyy"];
    
    //CGRect frame = CGRectMake(40, 100, 200, 21);
    
    //Stack View
    UIStackView *stackView = [[UIStackView alloc] init];
    
    stackView.axis = UILayoutConstraintAxisVertical;
    stackView.distribution = UIStackViewDistributionEqualSpacing;
    stackView.alignment = UIStackViewAlignmentCenter;
    stackView.spacing = 30;
    stackView.translatesAutoresizingMaskIntoConstraints = false;

    
    for (int i=0;i<5;i++)
    {
        UILabel *label = [[UILabel alloc] init];
        //Can't concat strings. immutable.
        [label setText: [NSString stringWithFormat:@"Today is: %@", [dateFormatter stringFromDate:[NSDate date]]]];
        [label setBackgroundColor:[UIColor clearColor]];
        [label setTextColor:[UIColor redColor]];
        //strings start with at. put f on float
        [label setFont:[UIFont fontWithName:@"Veranda" size:18.0f]];
        [label setShadowColor:[UIColor darkGrayColor]];
        
        UITapGestureRecognizer* gesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(userTappedOnLink:)];
        //let tap = UITapGestureRecognizer(target: self, action: #selector(ViewController.tapFunction))
        
        //label.isUserInteractionEnabled = true
        [label setUserInteractionEnabled:YES];
        [label addGestureRecognizer:gesture];
        label.tag = i;
        
        //[self.view addSubview:label];
        [stackView addArrangedSubview:label];
    }
    
    [self.view addSubview:stackView];
    
    
    //Layout for Stack View
    [stackView.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor].active = true;
    [stackView.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor].active = true;
    
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)userTappedOnLink:(UIGestureRecognizer*)gestureRecognizer
{
    printf("Ckicked on label\n");
    UIView *view = gestureRecognizer.view;
        switch (view.tag) {
            case 0:
                printf("0\n");
                break;
            case 1:
                printf("1\n");
                break;
            case 2:
                printf("2\n");
                break;
            case 3:
                printf("3\n");
                break;
            case 4:
                printf("4\n");
                break;
            default:
                break;
        }
}

@end

File: Day4/Autolayout Homework/Autolayout Homework/ViewController.swift
//
//  ViewController.swift
//  Autolayout Homework
//
//  Created by Consultant on 8/23/22.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        let date = Date()
        let calendar = Calendar.current
        let day = calendar.component(.weekday, from: date)
        let month = calendar.component(.month, from: date)
        let year = calendar.component(.year, from: date)
        
        //let frame = CGRect(x: 50, y: 100, width: 200, height: 90)
        
        let stackView   = UIStackView()
        stackView.axis  = NSLayoutConstraint.Axis.vertical
        stackView.distribution  = UIStackView.Distribution.equalSpacing
        stackView.alignment = UIStackView.Alignment.center
        stackView.spacing   = 16.0
        stackView.translatesAutoresizingMaskIntoConstraints = false
        
        
        
        for i in 0..<5
        {
            let label = UILabel()
            label.text = "Today is \(month)-\(day)-\(year)"
            label.backgroundColor = UIColor.clear
            label.textColor = UIColor.red
            label.font = UIFont(name:"Veranda", size: 18.0)
            label.shadowColor = UIColor.darkGray
            
            //Only last label is clickable?
            let tap = UITapGestureRecognizer(target: self, action: #selector(ViewController.tapFunction))
            label.isUserInteractionEnabled = true
            label.addGestureRecognizer(tap)
            label.tag = i
            
            //view.addSubview(label)
            stackView.addArrangedSubview(label)
        }
        
      
        
        view.addSubview(stackView)
        
        stackView.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true
        stackView.centerYAnchor.constraint(equalTo: self.view.centerYAnchor).isActive = true
        
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    @objc func tapFunction(sender: UITapGestureRecognizer) {
            print("tap working")
        let view:UIView = sender.view!
            switch (view.tag) {
            case 0:
                print(view.tag) //  info 0 tapped
            case 1:
                print(view.tag) //  info 1 tapped
            case 2:
                print(view.tag) //  info2 tapped
            case 3:
                print(view.tag) //  info 3 tapped
            case 4:
                print(view.tag) //  info 4 tapped
            default:
                break;
            }
        }
}

